chain/epoch_manager/src/lib.rs

```diff
@@ -154,6 +154,11 @@ impl EpochManager {
             // EpochId formula using T-2 for T=1, and height field is unused.
             let block_info = Arc::new(BlockInfo::default());
             let mut store_update = epoch_manager.store.store_update();
+            epoch_manager.save_epoch_start(
```

Comment:
add comment on why we need to save the epoch_start too
(as this is basically updating some columns in DB, right?)

Reply:
Oh, so I think this is not actually right. I'll need to revisit this on Monday, but it looks like the situation with epochs at genesis is quite curious. We have three blocks: 

```
dummy <- genesis <- first block
```

The dummy has hash 1111, genesis has hash 0xGENESIS and has dummy as its parent. 

Conceptually, `genesis` is in an epoch of its own. 

That is, first block starts a new epoch, with id 1111 (bc that's the last block of prev-prev epoch -- genessis). The epoch after the first one would would have id of 0xGENESIS. 

So, genesis and first block actually belong to different epochs.

And the problem is that we don't have a reasonable id for genesis' own epoch. The block info claims it is `1111`, but that's actually an id of the following epoch. 


That's why, when the chain has only genesis block,  when we try to get validator info for the genesis block, we end up trying to find first block of epoch 1111 which didn't actually happen yet.